Sparkle/SUUpdater.m

```diff
@@ -294,6 +299,24 @@ - (void)scheduleNextUpdateCheck
     self.checkTimer = [NSTimer scheduledTimerWithTimeInterval:delayUntilCheck target:self selector:@selector(checkForUpdatesInBackground) userInfo:nil repeats:NO]; // Timer is non-repeating, may have invalidated itself, so we had to retain it.
 }
 
+- (void)receiveSleepNote
+{
+    if (self.checkTimer)
+    {
+        [self.checkTimer invalidate];
```

Comment:
Is it possible that OS sends sleep, but not wake? Maybe it'd be safer to keep the timer during sleep, and reinit it in the wake event?

Reply:
thats a valid suspicion given the (lack) of reliability of many macOS APIs. however, we've used similar code for 6 years without problems. also for some months we are running similar code in a fashion that would send some feedback to our server if it were ever the case that those two notifications are not triggered in a pair. not a single hiccup to date. however, i can try to re-work the code to rely solely on the wake-note if you wish.

Reply:
If you say it's a well-tested method, let's roll with it.